spec/services/preserved_object_handler_update_version_spec.rb

```diff
@@ -340,4 +334,370 @@
+          expect(po).not_to have_received(:touch)
+          expect(pc).not_to have_received(:touch)
+        end
+
+        context 'incoming version newer than catalog versions (both) (happy path)' do
+          it 'updates PreservedCopy last_audited' do
+            orig_timestamp = pc.last_audited
+            sleep 1
+            po_handler.update_version_after_validation
+            expect(pc.reload.last_audited).to be > orig_timestamp
+          end
+          it 'updates PreservedCopy last_checked_on_storage' do
+            orig_timestamp = pc.last_checked_on_storage
+            po_handler.update_version_after_validation
+            expect(pc.reload.last_checked_on_storage).to be > orig_timestamp
+          end
+
+          it 'calls #update_online_version with validated = true and Status.invalid_moab' do
+            expect(po_handler).to receive(:update_online_version).with(true, Status.invalid_moab).and_call_original
+            po_handler.update_version_after_validation
+            skip 'test is lame b/c we only indirectly show the effects of #update_online_version in #update_version specs'
+          end
+
+          it 'updates PreservedCopy status to "moab_invalid" if it was "ok"' do
+            pc.status = Status.ok
+            pc.save!
+            po_handler.update_version_after_validation
+            expect(pc.reload.status).to eq Status.invalid_moab
+          end
+        end
+
+        RSpec.shared_examples 'unexpected version with validation' do |incoming_version|
+          let(:po_handler) { described_class.new(druid, incoming_version, 1, ep) }
+          let(:exp_msg_prefix) { "PreservedObjectHandler(#{druid}, #{incoming_version}, 1, #{ep})" }
+          let(:version_msg_prefix) { "#{exp_msg_prefix} incoming version (#{incoming_version})" }
+          let(:unexpected_version_msg) { "#{version_msg_prefix} has unexpected relationship to PreservedCopy db version; ERROR!" }
+
+          it "PreservedCopy version stays the same" do
+            pcv = pc.version
+            po_handler.update_version_after_validation
+            expect(pc.reload.version).to eq pcv
+          end
+          it "PreservedObject current_version stays the same" do
+            pocv = po.current_version
+            po_handler.update_version_after_validation
+            expect(po.reload.current_version).to eq pocv
+          end
+          it "PreservedCopy size stays the same" do
+            expect(pc.size).to eq 1
+            po_handler.update_version_after_validation
+            expect(pc.reload.size).to eq 1
+          end
+          it 'updates PreservedCopy last_audited field' do
+            orig_timestamp = pc.last_audited
+            sleep 1
+            po_handler.update_version_after_validation
+            expect(pc.reload.last_audited).to be > orig_timestamp
+          end
+          it 'updates PreservedCopy last_checked_on_storage' do
+            orig_timestamp = pc.last_checked_on_storage
```

Comment:
same (why no) `sleep` question

Reply:
same answer:  last_checked_on_storage is aware of fractions of a second and needs no sleep.  Comments added to sleep lines.